core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -532,26 +532,30 @@ static RelDataType deriveTypeSplit(SqlOperatorBinding operatorBinding,
   public static final SqlFunction EXTRACT_VALUE =
       SqlBasicFunction.create("EXTRACTVALUE",
           ReturnTypes.VARCHAR_2000.andThen(SqlTypeTransforms.FORCE_NULLABLE),
-          OperandTypes.STRING_STRING);
+          OperandTypes.STRING_STRING,
```

Comment:
![image](https://github.com/apache/calcite/assets/8057451/265fbf46-bb07-4910-8037-3f21226e97b7)

I find that EXTRACT_VALUE function would return the numeric value in some Scene.

I don't know whether it is appropriate to divide it into a String type function. I saw that it was divided into numeric function before.

Reply:
@LakeShen I get your point. Usually we classify functions according to the operands (in some cases, they can also be determined according to the return value type), which is not clearly specified by calcite. So we need to take a comprehensive view. These functions actually process XML. In database systems such as MySql or Oracle, they are divided into XML functions. The closest type in calcite is String.

Reply:
I have a meta-question: what is this field used for?
It doesn't seem to affect any of the tests.
Clearly, the meaning of this field is ambiguous too - it's not clear what the right choice is in all cases.
Can we just remove it completely?

Reply:
Hi, @mihaibudiu. thanks for reviewing. good question. 

Frankly speaking, `SqlFunctionCategory` is a very loose field now. In fact, there is no strict verification during lookup, so no test case covers it. However, it is currently widely used and cannot be deleted for subsequent expansion. We can refer to the comments of the current `lookupOperatorOverloads` and use it immediately when we need it.

https://github.com/apache/calcite/blob/597b53dbb075ab72700d40f66db81d2ac9c182b8/core/src/main/java/org/apache/calcite/sql/util/ReflectiveSqlOperatorTable.java#L94